sdk/search/azure-search/src/main/java/com/azure/search/models/DataSourceType.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Collection;
+
+/**
+ * Defines values for DataSourceType.
+ */
+public final class DataSourceType extends ExpandableStringEnum<DataSourceType> {
```

Comment:
Worth noting that whenever we get around to the preview SDK, there are a couple more types that should be included here.

Reply:
@careyjmac Are those types are already in the 2019-05-06-Preview Swagger spec?

Reply:
Not sure, Victor is the one who generally manages preview data sources.  Several of them have been preview for awhile from I'm pretty sure issues with the partner teams not cooperating with us, so they might never GA.  Worth a follow up as a backlog item at least.

Reply:
@careyjmac Please ensure this is tracked. We won't consider this blocking for this PR, since it covers only the GA API version.